tool/lib/license_utils.dart

```diff
@@ -0,0 +1,255 @@
+}
+
+class LicenseHeader {
+  /// If the [file] matches the given [existingLicenseText] within the first
+  /// number of [byteCount] bytes, return the 'existing_header' and
+  /// 'replacement_header' each with the stored value ('<value>'
+  /// if configured, defaults to [defaultStoreValue] or current year)
+  /// populated. For now, only up to one stored value is supported.
+  /// If the file can't be read or no match is found, throws an exception.
+  Future<Map<String, String>> getReplacementInfo(
+    File file,
+    String existingLicenseText,
+    String replacementLicenseText,
+    int byteCount, [
+    defaultStoredValue,
+  ]) async {
+    final stream = file
+        .openRead(0, byteCount)
+        .transform(utf8.decoder)
+        .handleError((e) => throw Exception(
+            'License header expected, but error reading file - $e',),);
+    await for (final content in stream) {
+      // Return just the license headers for the simple case with no stored
+      // value requested (i.e. content matches licenseText verbatim)
+      if (content.contains(existingLicenseText)) {
+        final String storedName = _parseStoredName(replacementLicenseText);
+        replacementLicenseText = replacementLicenseText.replaceAll(
+          '<$storedName>',
+          defaultStoredValue ?? DateTime.now().year.toString(),
+        );
+        return {
+          existingHeaderKey: existingLicenseText,
+          replacementHeaderKey: replacementLicenseText,
+        };
+      }
+      // Return a non-empty map for the case where there is a stored value
+      // requested (i.e. when there is a '<value>' defined in the license text)
+      final String storedName = _parseStoredName(existingLicenseText);
+      if (storedName.isNotEmpty) {
+        return _processHeaders(
+          storedName,
+          existingLicenseText,
+          replacementLicenseText,
+          content,
+        );
+      }
+    }
+    throw Exception('License header expected in ${file.path}, but not found!');
+  }
+
+  /// Returns a copy of the given [file] with the [existingHeader] replaced by
+  /// the [replacementHeader]. Reads and writes the entire file contents all
+  /// at once, so performance may degrade for large files.
+  Future<File> rewriteLicenseHeader(
+    File file,
+    String existingHeader,
+    String replacementHeader,
+  ) async {
+    final File rewrittenFile = File('${file.path}.tmp');
+    // TODO: [mossmana] Convert to using a stream
+    await file.readAsString().then((String contents) async {
+      final String replacementContents = contents.replaceFirst(
+        existingHeader,
+        replacementHeader,
+      );
+      await rewrittenFile.writeAsString(replacementContents);
+    });
+    return rewrittenFile;
+  }
+
+  /// Bulk update license headers for files in the [directory] as configured
+  /// in the [config]. Returns a list of file paths that were updated.
+  Future<Map<String, List<String>>> bulkUpdate(
```

Comment:
does this need to be a method on the LicenseHeader class? or should it be a standalone method in the file? Another option is to make this a static method so that an instance of LicenseHeader does not need to be created in order to call this method.

Reply:
It references class member functions such as `getReplacementInfo` and `rewriteLicenseHeader`.